### PR TITLE
Join an existing shopping list

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person
## Summary
We want to allow users to join existing shopping lists so they can manage them with friends or family. This would take the following steps:
1. User enters an existing token into a form
2. The token is saved to `localStorage`
3. The user is taken to the `List` view and shown the existing list.

## Acceptance criteria
If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- Be sure to use the `setListToken` function to save the token that the user enters!
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
- If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue [3](https://github.com/the-collab-lab/tcl-49-smart-shopping-list/issues/3)).

### PR DESCRIPTION
**Pull request by Mama-simba**
## Description

Our main goal is to allow the users to join an existing list. If the users have a token, they will be redirected straight away to the shopping list view.
If they do not have any token (the list doesn't exist), they will be redirected to the Home view and will see an error message that explains the problem.

## Related Issue

Closes #5 

## Acceptance Criteria

If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

Watch out the this demo that displays the join existing list functionality:

[joinshoppinglist.webm](https://user-images.githubusercontent.com/68116083/196958044-edce04f6-8de7-4973-a1bf-395ec4b84a7b.webm)

## Testing Steps / QA Criteria

You can test it out here: https://tcl-49-smart-shopping-list--pr22-cp-ve-join-existing-2asb3zk6.web.app/

### CODE REVIEW

**Comment by Terieyenike** [comment]
Tested the app, and so far, it is working. 
One quick question, how do I, as a first-time user, know a token to use to validate me to see the list?

**Comment by mohanner-coder** [comment]
> Tested the app, and so far, it is working. One quick question, how do I, as a first-time user, know a token to use to validate me to see the list?

I think it isn't part of the task, but we will need to get Golfo to explain how to go about this. What we did to know if our create list component is functioning was to delete the token which we already had from the last project in the console. You will notice that the create list component will appear when you have no token on your console.

**Comment by Mama-simba** [comment]
As you have probably already seen, so far either you get a token automatically or you don't get anything but can try with the default one from Firebase (my test list)
We have been checking the tokens on the browser (localStorage) as we show in the demo.

**Comment by Terieyenike** [comment]
Done with the code review, and it was indeed a lovely job you did.
Keep up the excellent work.

File: src/views/List.jsx [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+export function List({ listToken }) {
+	const [data, setData] = useState([]);
+	const [loading, setLoading] = useState(true);
+
+	useEffect(() => {
+		streamListItems(listToken, (snapshot) => {
+			// 		/**
+			// 		 * Read the documents in the snapshot and do some work
+			// 		 * on them, so we can save them in our React state.
+			// 		 *
```
Comment (Terieyenike) [inline src/views/List.jsx]: remove comments from this component.

**Review by Terieyenike — APPROVED** [review]
the app works for me, and I confirmed the edge cases.
Great job, guys.

File: src/views/List.jsx (hunk 1 above)
Comment (Mama-simba) [reply src/views/List.jsx]: Done!

File: src/App.jsx [hunk 2]
```diff
@@ -2,58 +2,50 @@
...
+	const [displayName, setDisplayName] = useState('');
 	const [listToken, setListToken] = useStateWithStorage(
 		null,
 		'tcl-shopping-list-token',
 	);
 
+	useEffect(() => {
+		if (listToken) navigate('/list');
+
+		return;
```
Comment (pmbanugo) [inline src/App.jsx]:
```suggestion
```

This could be skipped and it'll behave the same way

**Review by pmbanugo — APPROVED** [review]
Awesome teamwork! The PR description has improved 👍🏽 

I left one minor comment that you could resolve and you can merge when you're ready

**Comment by mohanner-coder** [comment]
> Done with the code review, and it was indeed a lovely job you did. Keep up the excellent work.

Thanks  Teri

File: src/App.jsx (hunk 2 above)
Comment (mohanner-coder) [reply src/App.jsx]: Thanks Peter. We've removed it.

**Pull request merged**